### PR TITLE
bigquery: add snippet for bigquery_update_dataset_access

### PR DESCRIPTION
**Pull request by shollyman**
Demonstrates adding an ACE to a dataset ACL

### CODE REVIEW

File: bigquery/snippets/snippet_test.go [hunk 1]
```diff
@@ -48,6 +48,10 @@ func TestAll(t *testing.T) {
 		t.Errorf("failed to create dataset: %v", err)
 	}
 
+	if err := updateDatasetAccessControl(client, datasetID); err != nil {
+		t.Errorf("Failed to update access control for dataset: %v", err)
```
Comment (broady) [inline bigquery/snippets/snippet_test.go]: nit: lowercase f to be consistent.

but really the style can simply be:

`t.Errorf("updateDataSetAccessControl: %v", err)`

And you can add the datasetID if it's relevant.

`t.Errorf("updateDataSetAccessControl(%q): %v", datasetID, err)`

Comment (shollyman) [reply bigquery/snippets/snippet_test.go]: If that's the dominant idiom, should I go ahead and plumb the method name into all the checks?

Comment (broady) [reply bigquery/snippets/snippet_test.go]: if you're up for it! I'm happy to come and clean it up, too, if you don't have time.

**Comment by broady** [comment]
nice! LGTM.

/cc @tbpg

**Pull request merged**